API/Services/TachiyomiService.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using API.DTOs;
+using System.Threading.Tasks;
+using API.Data;
+using System.Collections.Immutable;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using API.Comparators;
+using API.Entities;
+using AutoMapper;
+namespace API.Services;
+
+public interface ITachiyomiService
+{
+    Task<ChapterDto> GetLatestChapter(int seriesId, int userId);
+    Task<bool> MarkChaptersUntilAsRead(AppUser userWithProgress, int seriesId, float chapterNumber);
+}
+
+
+public class TachiyomiService : ITachiyomiService
+{
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly IReaderService _readerService;
+    private readonly IMapper _mapper;
+    private readonly CultureInfo _englishCulture = CultureInfo.CreateSpecificCulture("en-US");
+
+    public TachiyomiService(IUnitOfWork unitOfWork,IMapper mapper,IReaderService readerService)
+    {
+        _unitOfWork = unitOfWork;
+        _readerService = readerService;
+        _mapper = mapper;
+    }
+
+    public async Task<ChapterDto> GetLatestChapter(int seriesId,int userId)
+//    public async Task<ChapterDto> GetLatestChapter(int seriesId, int userId)
+    {
+
+
+        var currentChapter = await _readerService.GetContinuePoint(seriesId, userId);
+
+        var prevChapterId =
+            await _readerService.GetPrevChapterIdAsync(seriesId, currentChapter.VolumeId, currentChapter.Id, userId);
+
+        // If prevChapterId is -1, this means either nothing is read or everything is read.
+        if (prevChapterId == -1)
+        {
+            var series = await _unitOfWork.SeriesRepository.GetSeriesDtoByIdAsync(seriesId, userId);
+            var userHasProgress = series.PagesRead != 0 && series.PagesRead <= series.Pages;
+
+            // If the user doesn't have progress, then return null, which the extension will catch as 204 (no content) and report nothing as read
+            if (!userHasProgress) return null;
+
+            // Else return the max chapter to Tachiyomi so it can consider everything read
+            var volumes = (await _unitOfWork.VolumeRepository.GetVolumes(seriesId)).ToImmutableList();
+            var looseLeafChapterVolume = volumes.FirstOrDefault(v => v.Number == 0);
+            if (looseLeafChapterVolume == null)
+            {
+                var volumeChapter = _mapper.Map<ChapterDto>(volumes.Last().Chapters.OrderBy(c => float.Parse(c.Number), ChapterSortComparerZeroFirst.Default).Last());
+                if (volumeChapter.Number == "0")
+                {
+                    var volume = volumes.First(v => v.Id == volumeChapter.VolumeId);
+                    return new ChapterDto()
+                    {
+                        Number = (volume.Number / 10000f).ToString("R", _englishCulture)
+                    };
+                }
+
+                return new ChapterDto()
+                {
+                    Number = (int.Parse(volumeChapter.Number) / 10000f).ToString("R", _englishCulture)
```

Comment:
Add a <remarks> in the method documentation to explain how we handle the / 10000. 

Also change 10000f -> 10_000f